### PR TITLE
Update codedeploy as of 09/20/18

### PR DESCRIPTION
**Pull request by mprince**
[AWS CloudFormation Release History](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html)

### CODE REVIEW

File: troposphere/codedeploy.py [hunk 1]
```diff
@@ -177,10 +178,28 @@ class DeploymentGroup(AWSObject):
         'DeploymentGroupName': (basestring, False),
         'DeploymentStyle': (DeploymentStyle, False),
         'Ec2TagFilters': ([Ec2TagFilters], False),
+        'Ec2TagSet': ([Ec2TagFilters], False),
```
Comment (phobologic) [inline troposphere/codedeploy.py]: Hey @mprince, I believe this requires a new Property type:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codedeploy-deploymentgroup-ec2tagset.html

File: troposphere/codedeploy.py [hunk 2]
```diff
@@ -177,10 +178,28 @@ class DeploymentGroup(AWSObject):
...
         'LoadBalancerInfo': (LoadBalancerInfo, False),
         'OnPremisesInstanceTagFilters': (
             [OnPremisesInstanceTagFilters], False
         ),
+        'OnPremisesInstanceTagSet': (
+            [OnPremisesInstanceTagFilters], False
```
Comment (phobologic) [inline troposphere/codedeploy.py]: Same here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codedeploy-deploymentgroup-onpremisestagset.html

**Comment by mprince** [comment]
@phobologic doc is kind of confusing in the naming convention with `Set` and `SetObject`.

**Comment by phobologic** [comment]
@mprince yeah, the Codebuild one was even more annoying - they introduce 3 different properties for TargetSettings or whatever, and 2 of the 3 had the same properties.  Sometimes I feel like the CF teams are just trying to make this stuff hard :)  Thanks for this, I'll dig in later today.

File: troposphere/codedeploy.py [hunk 3]
```diff
@@ -164,6 +173,30 @@ class TriggerConfig(AWSProperty):
...
+
+class OnPremisesTagGroup(AWSProperty):
+    props = {
+        'OnPremisesTagGroup': ([TagFilters], False)
+    }
+
+
+class OnPremisesTagSet(AWSProperty):
+    props = {
+        'OnPremisesTagSet': ([OnPremisesTagGroup], False)
```
Comment (phobologic) [inline troposphere/codedeploy.py]: This property should be `OnPremisesTagSetList` it looks like:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codedeploy-deploymentgroup-onpremisestagset.html

It looks like that's supposed to be a list of  `OnPremisesTagSetListObject` objects:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codedeploy-deploymentgroup-onpremisestagsetlistobject.html

File: troposphere/codedeploy.py [hunk 4]
```diff
@@ -164,6 +173,30 @@ class TriggerConfig(AWSProperty):
...
 
+class Ec2TagGroup(AWSProperty):
+    props = {
+        'Ec2TargetGroup': ([Ec2TagFilters], False)
+    }
+
+
+class Ec2TagSet(AWSProperty):
+    props = {
+        'Ec2TagSet': ([Ec2TagGroup], False)
```
Comment (phobologic) [inline troposphere/codedeploy.py]: This looks like the property is `EC2TagSetList`, which is a list of `EC2TagSetListObject` objects,

**Pull request merged**